azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/TrafficManagerEndpoint.java

```diff
@@ -0,0 +1,500 @@
+             * Specify that the endpoint is a nested traffic manager profile.
+             *
+             * @param nestedProfile the nested traffic manager profile
+             * @return the next stage of the endpoint definition
+             */
+            WithEndpointThreshold<ParentT> withNestedProfile(TrafficManagerProfile nestedProfile);
+        }
+
+        /**
+         * The stage of the nested traffic manager profile endpoint definition allowing to specify the minimum
+         * endpoints to be online in the nested profile to consider it as not degraded.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithEndpointThreshold<ParentT> {
+            /**
+             * Specifies the child endpoint threshold.
+             *
+             * @param count the number of child endpoints to be online to consider nested profile as healthy
+             * @return the next stage of the endpoint definition
+             */
+            WithSourceTrafficLocation<ParentT> withMinimumChildEndpoints(int count);
+        }
+
+        /**
+         * The stage of the traffic manager endpoint definition allowing to specify the location of the external
+         * or nested profile endpoints.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithSourceTrafficLocation<ParentT> {
+            /**
+             * Specifies the location of the endpoint that will be used when the parent profile is configured with
+             * Performance routing method {@link TrafficRoutingMethod#PERFORMANCE}.
+             *
+             * @param location the location
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT> withSourceTrafficLocation(Region location);
+        }
+
+        /**
+         * The stage of the traffic manager endpoint definition allowing to specify the endpoint weight.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithRoutingWeight<ParentT> {
+            /**
+             * Specifies the weight for the endpoint that will be used when the parent profile is configured with
+             * Weighted routing method {@link TrafficRoutingMethod#WEIGHTED}.
+             *
+             * @param weight the endpoint weight
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT> withRoutingWeight(int weight);
+        }
+
+        /**
+         * The stage of the traffic manager endpoint definition allowing to specify the endpoint priority.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithRoutingPriority<ParentT> {
+            /**
+             * Specifies the priority for the endpoint that will be used when the parent profile is configured with
```

Comment:
similar to before: "Specifies the priority for the endpoint that will be used when the priority-based routing method is enabled on the profile."